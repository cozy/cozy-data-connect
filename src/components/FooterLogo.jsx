import React from 'react'
import PropTypes from 'prop-types'
import get from 'lodash/get'
import { withClient } from 'cozy-client'

const fetchHomeLogos = async client => {
  try {
    const rootURL = client.getStackClient().uri
    const context = await client
      .getStackClient()
      .fetchJSON('GET', '/settings/context')
    const logos = get(context, 'data.attributes.home_logos', {})

    return Object.keys(logos).reduce((acc, logoSrc) => {
      acc[`${rootURL}/assets${logoSrc}`] = logos[logoSrc]
      return acc
    }, {})
  } catch (error) {
    return {}
  }
}

export class FooterLogo extends React.Component {
  state = {
    logos: {}
  }

  async componentDidMount() {
    const logos = await fetchHomeLogos(this.props.client)
    this.setState({ logos })
  }

  render() {
    const { logos } = this.state
    return Object.keys(logos).length === 0 ? (
      false
    ) : (
      <div className="u-maw-7 u-mh-auto u-mt-1 u-pv-1 u-flex u-flex-row u-flex-items-center u-flex-justify-center u-flex-wrap">
        {Object.entries(logos).map(([logoSrc, logoAlt]) => (
          <img
            key={logoSrc}
            src={logoSrc}
            alt={logoAlt}
            className="u-ph-1 u-pv-half u-mah-5"
          />
        ))}
      </div>
    )
  }
}

PropTypes.propTypes = {
  client: PropTypes.object
}

export default withClient(FooterLogo)
