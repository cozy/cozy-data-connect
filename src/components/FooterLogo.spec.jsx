import React from 'react'
import { shallow } from 'enzyme'
import { FooterLogo } from './FooterLogo'

describe('FooterLogo', () => {
  const mockClient = {
    getStackClient: () => mockClient,
    uri: 'http://cozy.example.com',
    fetchJSON: jest.fn()
  }

  it('should render nothing when there are no logos', () => {
    mockClient.fetchJSON.mockResolvedValue({
      data: {
        attributes: {}
      }
    })
    const component = shallow(<FooterLogo client={mockClient} />)
    expect(component.getElement()).toEqual(null)
  })

  it('should render multiple logos', async () => {
    mockClient.fetchJSON.mockResolvedValue({
      data: {
        attributes: {
          home_logos: {
            '/path1/cozy.svg': 'alt text 1',
            '/path/2/cozy-with_complex-name.svg': 'alt text 2'
          }
        }
      }
    })
    const component = shallow(<FooterLogo client={mockClient} />)
    await new Promise(resolve => setImmediate(resolve)) // await the didMount
    expect(component.getElement()).toMatchSnapshot()
  })
})
