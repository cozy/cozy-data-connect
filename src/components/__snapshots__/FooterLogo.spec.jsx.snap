// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLogo should render multiple logos 1`] = `
<div
  className="u-maw-7 u-mh-auto u-mt-1 u-pv-1 u-flex u-flex-row u-flex-items-center u-flex-justify-center u-flex-wrap"
>
  <img
    alt="alt text 1"
    className="u-ph-1 u-pv-half u-mah-5"
    src="http://cozy.example.com/assets/path1/cozy.svg"
  />
  <img
    alt="alt text 2"
    className="u-ph-1 u-pv-half u-mah-5"
    src="http://cozy.example.com/assets/path/2/cozy-with_complex-name.svg"
  />
</div>
`;
